packages/tpetra/core/src/Tpetra_Details_normImpl.hpp

```diff
@@ -333,6 +334,15 @@ normImpl (MagnitudeType norms[],
 
   Impl::lclNormImpl (normsOut, X, numVecs, whichVecs,
                      isConstantStride, whichNorm);
+
+  // lbv 03/15/23: the data from the local norm calculation
+  // better really be available before communication happens
+  // so fencing to make sure the local computations have
+  // completed on device. We might want to make this an
+  // execution space fence down the road?
+  execution_space exec_space_instance = execution_space();
+  exec_space_instance.fence();
```

Comment:
Note you better be sure that the default instance is what is used in lclNormImpl. Also presumably that call is only asynchronous because it calls a parallel reduce with a Kokkos::View as output argument instead of a scalar?
parallel_reduce with a scalar output argument is blocking, with View output arguments its not.